### DISCUSSION TITLE
Custom rule to allow email auth

### DISCUSSION THREAD

**Discussion by Falci** [supabase/supabase#18139] [Questions] [answered]
Hi all,
I'm planning to use Supabase's auth in a new project. It could be email or oauth.
My project will sync with a 3rd-party app that doesn't support oauth.
They have a simple API, and I can retrieve a user by their email.

Can I restrict the sign up in Supabase to only allows email that are already in use this 3rd-party app?
It's fine if I need to create my own API in between

**Comment by GaryAustin1** [comment, marked as answer]
You can use a trigger function on auth.users and compare the email to your list of emails and error.  There is something like a 3 second timeout in the auth database call so your api needs to respond quickly.  You would use http extension to call the API.

**Reply by Falci** [reply]
Sounds interesting. Could you point to some sample code?

**Reply by GaryAustin1** [reply]
Sorry, no.  I've seen users use http calls in the auth.trigger function to APIs, lots of http extension stuff out there, and the user management starter (like in the SQL editor Quickstarts) for the function and trigger.  But can't point you to a total solution.
